### PR TITLE
Add profit output for clusters

### PR DESCRIPTION
**Pull request by guilpier-code**

### CODE REVIEW

File: src/solver/variable/economy/profitByPlant.h [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        void hourForEachThermalCluster(State& state, unsigned int numSpace)
+        {
+            // Useful local variables
+            double* MarginalCosts = state.hourlyResults->CoutsMarginauxHoraires;
+            auto* cluster = state.thermalCluster;
+            double hourlyClusterProduction = state.thermalClusterProduction;
+            uint hourInTheWeek = state.hourInTheWeek;
+            uint hourInTheYear = state.hourInTheYear;
+            int modulationCostColumn = Data::thermalModulationCost;
+
```
Comment (flomnes) [inline src/solver/variable/economy/profitByPlant.h]: Not really necessary... You could use `Data::thermalModulationCost` at line 328.

File: src/solver/variable/economy/profitByPlant.h [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        //! Unit
+        static const char* Unit()
+        {
+            return "Profit - Euro";
+        }
+
+        //! The short description of the variable
+        static const char* Description()
+        {
+            return "Profit costs by all the clusters";
```
Comment (flomnes) [inline src/solver/variable/economy/profitByPlant.h]: Profit for all thermal units

File: src/solver/variable/economy/profitByPlant.h [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    {
+        //! Caption
+        static const char* Caption()
+        {
+            return "Profit by plant";
+        }
+        //! Unit
+        static const char* Unit()
+        {
+            return "Profit - Euro";
```
Comment (flomnes) [inline src/solver/variable/economy/profitByPlant.h]:
```suggestion
            return "Euro";
```

**Review by flomnes — COMMENTED** [review]
I tested this manually, and got slightly strange results
Study : **007 Thermal fleet - Fast unit commitment/**
Hour **34**
Cluster **base**
Manual computation : 71874036.00
Profit computed by Antares : 71874032

I have a difference of 4 EUR, it is probably caused by rounding errors.

File: src/solver/variable/economy/profitByPlant.h (hunk 3 above)
Comment (guilpier-code) [reply src/solver/variable/economy/profitByPlant.h]: ok, done.

File: src/solver/variable/economy/profitByPlant.h (hunk 2 above)
Comment (guilpier-code) [reply src/solver/variable/economy/profitByPlant.h]: ok, done.

File: src/solver/variable/economy/profitByPlant.h (hunk 1 above)
Comment (guilpier-code) [reply src/solver/variable/economy/profitByPlant.h]: ok, done.

**Review by flomnes — APPROVED** [review]

**Pull request merged**